sdk/src/azure/core/az_json_token.c

```diff
@@ -325,6 +325,52 @@ AZ_NODISCARD static az_result _az_json_token_get_string_helper(
   return AZ_OK;
 }
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
+    int32_t destination_max_size,
+    int32_t* out_string_length)
+{
+  _az_PRECONDITION_VALID_SPAN(json_string, 1, false);
+  _az_PRECONDITION_NOT_NULL(destination);
+  _az_PRECONDITION(destination_max_size > 0);
+
+  int32_t position = 0;
+  for (int32_t i = 0; i < json_string._internal.size; i++)
+  {
+    uint8_t current_char;
+    current_char = json_string._internal.ptr[i];
+    if (current_char == '\\' && i < json_string._internal.size)
+    {
+      uint8_t next_char = json_string._internal.ptr[i + 1];
```

Comment:
Looking at test coverage, I just noticed, this `i + 1` can end up accessing the address outside the bounds of the span. Please fix.

Let's say input is 1 character long: `\`

current_char = `\` and i = 0, which is less than the length (1).
next_char = json_string._internal.ptr[1]; // undefined behavior, accessing out-of-memory

Reply:
there is a test for that line 3087